### ISSUE TITLE
Getting different results between eslint-webpack-plugin versions

### ISSUE THREAD

**Issue by Buyooo** [webpack-contrib/eslint-webpack-plugin#103]
- Operating System: Win 10
- Node Version: 14.5.1
- NPM Version: 6.14.9
- webpack Version: 5.39.0
- eslint-webpack-plugin Version: 2.5.4

### Expected Behavior

Same behaviour as eslint-webpack-plugin Version: 2.1.0

### Actual Behavior

When running webpack it builds without flagging the interface `INotUsed` on the `testFile.tsx`

### Code

```js
const path = require('path');
const ESLintPlugin = require('eslint-webpack-plugin');

module.exports = {
    mode: 'development',
    entry: './src/index.tsx',
    devtool: "inline-source-map",
    output: {
        filename: 'main.js',
        path: path.resolve(__dirname, 'dist'),
    },
    resolve: {
        // Add `.ts` and `.tsx` as a resolvable extension.
        extensions: [".ts", ".tsx", ".js"]
    },
    module: {
        rules: [
            // all files with a `.ts` or `.tsx` extension will be handled by `ts-loader`
            {
                test: /\.tsx?$/,
                loader: "ts-loader"
            }
        ]
    },
    plugins: [
        new ESLintPlugin({  
            files: './src/**/*.*',            
            extensions: ['.js', '.ts', '.tsx'],
            failOnWarning: true
        }),
    ],
};
```

### How Do We Reproduce?
1.  Clone small repo:
https://github.com/Buyooo/ESLint-test
2. Run `npm run lint` then `npm run build` (both report the same error)
3. Update `eslint-webpack-plugin` to `latest` (2.5.4 as of this writing)
4. Run same commands in step 2 again

**Comment by ricardogobbosouza** [comment]
Hi @Buyooo 
In version [v2.2.0](https://github.com/webpack-contrib/eslint-webpack-plugin/releases/tag/v2.2.0) we only use the import graph. We do not linting all files.
You can create a script to do this, like you did at https://github.com/Buyooo/ESLint-test/blob/main/package.json#L9

**Comment by Buyooo** [comment]
@ricardogobbosouza Thanks for the explanation.
